salt/modules/reg.py

```diff
@@ -747,3 +747,56 @@ def delete_value(hive, key, vname=None, use_32bit_registry=False):
         log.error('ValueName: {0}'.format(local_vname))
         log.error('32bit Reg: {0}'.format(use_32bit_registry))
         return False
+
+
+def regfile_import(source, use_32bit_registry=False):
+    '''
+    Import registry settings from a Windows Windows ``REG`` file by invoking ``REG.EXE``.
+
+    .. versionadded:: Nitrogen
+
+    Usage:
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt machine1 reg.regfile_import salt://win/printer_config/110_Canon/postinstall_config.reg
+
+    :param str source: The full path of the ``REG`` file. This
+        can be either a local file path or a URL type supported by salt
+        (e.g. ``salt://salt_master_path``).
+
+    :param bool use_32bit_registry: If the value of this paramater is ``True``
+        then the ``REG`` file will be imported into the Windows 32 bit registry.
+        Otherwise the Windows 64 bit registry will be used.
+
+    :return: If the value of ``source`` is an invalid path or otherwise
+       causes ``cp.cache_file`` to return ``False`` then
+       the function will not return and
+       a ``ValueError`` exception will be raised.
+       If ``reg.exe`` exits with a non-0 exit code, then
+       a ``CommandExecutionError`` exception will be
+       raised. On success this function will return
+       ``True``.
+
+    :rtype: bool
+
+    '''
+    cache_path = __salt__['cp.cache_file'](source)
+    if not cache_path:
+        error_msg = "File/URL '{0}' probably invalid.".format(source)
+        raise ValueError(error_msg)
+    if use_32bit_registry:
+        word_sz_txt = "32"
+    else:
+        word_sz_txt = "64"
+    cmd = 'reg import "{0}" /reg:{1}'.format(cache_path, word_sz_txt)
```

Comment:
Since we're importing a reg file, the only Unicode issues I foresee would be with the name of the file itself. This shouldn't be as big a problem in Py3.